tensorflow/compiler/mlir/xla/tests/hlo-legalize-to-lhlo.mlir

```diff
@@ -432,7 +432,23 @@ func @dot(%arg0: tensor<1024x1024xf32>) -> tensor<1024x1024xf32> {
 // CHECK-SAME: (%[[ARG0:.*]]: [[TYPE:.*]],
 // CHECK-SAME:  %[[RESULT:.*]]: [[TYPE]])
 // CHECK: "xla_lhlo.dot"(%[[ARG0]], %[[ARG0]], %{{.*}}) : ([[TYPE]], [[TYPE]], [[TYPE]]) -> ()
-    %dot = "xla_hlo.dot"(%arg0, %arg0)
-      : (tensor<1024x1024xf32>, tensor<1024x1024xf32>) -> tensor<1024x1024xf32>
-    return %dot : tensor<1024x1024xf32>
-  }
+  %dot = "xla_hlo.dot"(%arg0, %arg0)
+     : (tensor<1024x1024xf32>, tensor<1024x1024xf32>) -> tensor<1024x1024xf32>
+  return %dot : tensor<1024x1024xf32>
+}
+
+// -----
+
+// Test whether std.constant's generating tensors are converted to lhlo.const.
+func @std.const.tensor() {
+  %cst = constant  dense<0.000000e+00> : tensor<f32>
```

Comment:
why is it not an `xla_hlo.constant`?

Reply:
> why is it not an `xla_hlo.constant`?

Please see here: https://github.com/tensorflow/tensorflow/issues/39895